flexget/plugins/list/radarr_list.py

```diff
@@ -0,0 +1,593 @@
+    """ Makes a PUT request and returns the JSON response """
+    try:
+        response = requests.put(url, headers=headers)
+        if response.status_code == 200:
+            return response.json()
+        else:
+            raise RadarrRequestError(
+                "Invalid response received from Radarr: %s" % response.content
+            )
+    except RequestException as e:
+        raise RadarrRequestError("Unable to connect to Radarr at %s. Error: %s" % (url, e))
+
+
+class RadarrAPIService:
+    """ Handles all communication with the Radarr REST API """
+
+    def __init__(self, api_key, base_url, port=None):
+        self.api_key = api_key
+        parsed_base_url = urlparse(base_url)
+
+        if parsed_base_url.port:
+            port = int(parsed_base_url.port)
+
+        self.api_url = "%s://%s:%s%s/api/" % (
+            parsed_base_url.scheme,
+            parsed_base_url.netloc,
+            port,
+            parsed_base_url.path,
+        )
+
+    def get_profiles(self):
+        """ Gets all profiles """
+        request_url = self.api_url + "profile"
+        headers = self._default_headers()
+        json_response = request_get_json(request_url, headers)
+        return json_response
+
+    def get_movies(self):
+        """ Gets all movies """
+        request_url = self.api_url + "movie"
+        headers = self._default_headers()
+        json_response = request_get_json(request_url, headers)
+        return json_response
+
+    def get_root_folders(self):
+        """ Gets the root folders """
+        request_url = self.api_url + "rootfolder"
+        headers = self._default_headers()
+        json_response = request_get_json(request_url, headers)
+        return json_response
+
+    def delete_movie(self, movie_id):
+        """ Deletes a movie provided by its id """
+        request_url = self.api_url + "movie/" + str(movie_id)
+        headers = self._default_headers()
+        json_response = request_delete_json(request_url, headers)
+        return json_response
+
+    def lookup_by_term(self, term):
+        """ Returns all movies that matches the search term """
+        term = quote(term)
+        request_url = self.api_url + "movies/lookup?term=" + term
+        headers = self._default_headers()
+        json_response = request_get_json(request_url, headers)
+        return json_response
+
+    def lookup_by_imdb(self, imdb_id):
+        """ Returns all movies that matches the imdb id """
+        # TODO: make regexp check that imdb_id really is an IMDB_ID
+        request_url = self.api_url + "movies/lookup/imdb?imdbId=" + imdb_id
+        headers = self._default_headers()
+        json_response = request_get_json(request_url, headers)
+        return json_response
+
+    def lookup_by_tmdb(self, tmdb_id):
+        """ Returns all movies that matches the tmdb id """
+        tmdb_id = int(tmdb_id)
+        request_url = self.api_url + "movies/lookup/tmdb?tmdbId=" + str(tmdb_id)
```

Comment:
This is a little weird. int then str? Why? Remove leading zeros?

Reply:
Yeah, dunno. I just added str(..) around it as ide was complaining about it